components/hab/src/main.rs

```diff
@@ -109,7 +109,11 @@ fn main() {
     env_logger::init();
     let mut ui = UI::default_with_env();
     let flags = FeatureFlag::from_env(&mut ui);
-    thread::spawn(analytics::instrument_subcommand);
+
+    if analytics::analytics_enabled() {
+        thread::spawn(analytics::instrument_subcommand);
+    }
```

Comment:
Seems like if we're doing the check on the outside we shouldn't [*also* do the check in the inside](https://github.com/habitat-sh/habitat/blob/master/components/hab/src/analytics.rs#L235-L238).

Reply:
Happy to make that change, I left it to protect against callers who called it without checking.  But since calls to that function are minimal (this is the only one) I suppose that isn't much of a risk.